doc/manual/src/glossary.md

```diff
@@ -3,17 +3,24 @@
 - [derivation]{#gloss-derivation}
 
   A description of a build task. The result of a derivation is a
-  store object. Derivations are typically specified in Nix expressions
+  store object. Derivations created by Nix expressions are specified
   using the [`derivation` primitive](./language/derivations.md). These are
   translated into low-level *store derivations* (implicitly by
-  `nix-env` and `nix-build`, or explicitly by `nix-instantiate`).
+  `nix-build`, or explicitly by `nix-instantiate`).
+
+  Package definitions, such as in [nixpkgs], are written using one or
+  more derivations, typically with the use of helper functions such as `mkDerivation`.
 
   [derivation]: #gloss-derivation
+  [nixpkgs]: https://github.com/NixOS/nixpkgs
 
 - [store derivation]{#gloss-store-derivation}
 
   A [derivation] represented as a `.drv` file in the [store].
-  It has a [store path], like any [store object].
+  It has a [store path], like any [store object]. It is the
+  [instantiated][instantiate] form of a derivation. These are built
+  into *outputs* (implicitly by `nix-build`, or explicitly
+  by `nix-store --realise`).
```

Comment:
Maybe you mean that store derivations are the basis for realisation? Adding something like that could work out. But for the sake of merging quickly, this can be a follow-up PR.